core/drand_daemon_interceptors.go

```diff
@@ -32,10 +33,20 @@ func (dd *DrandDaemon) NodeVersionValidator(ctx context.Context, req interface{}
 		return handler(ctx, req)
 	}
 
-	rcvVer := commonutils.Version{Major: v.Major, Minor: v.Minor, Patch: v.Patch}
+	prerelease := ""
+	if v.Prerelease != nil {
+		prerelease = *v.Prerelease
+	}
+	rcvVer := commonutils.Version{
+		Major:      v.Major,
+		Minor:      v.Minor,
+		Patch:      v.Patch,
+		Prerelease: prerelease,
+	}
 	if !dd.version.IsCompatible(rcvVer) {
 		dd.log.Warnw("", "node_version_interceptor", "node version rcv is no compatible --> rejecting message", "version", rcvVer)
-		return nil, status.Error(codes.PermissionDenied, "Node Version not valid")
+		msg := fmt.Sprintf("Incompatible node version. Current: %s, received: %s", dd.version, rcvVer)
```

Comment:
Does that print properly the version? I would have expected a `%v` instead of a `%s`.
I'll try on playground :P

Reply:
Yeah, it doesn't, I think:
https://go.dev/play/p/S7542GoqEv1

Reply:
done, and sorry about that